directus-aws/index.md

```diff
@@ -0,0 +1,206 @@
+version: "3"
+services:
+  directus:
+    image: directus/directus:10.8.3
+    ports:
+      - 80:80
+    volumes:
+      - ./uploads:/directus/uploads
+      - ./extensions:/directus/extensions
+    environment:
+      PORT: 80
+      KEY: "replace-with-random-value"
+      SECRET: "replace-with-random-value"
+      ADMIN_EMAIL: "admin@example.com"
+      ADMIN_PASSWORD: "d1r3ctu5"
+      DB_CLIENT: "pg"
+      DB_HOST: "YOUR_RDS_DB_URL"
+      DB_PORT: 5432
+      DB_DATABASE: "postgres"
+      DB_USER: "YOUR_DB_USER"
+      DB_PASSWORD: "YOUR_DB_PASSWORD"
+      DB_SSL__REJECT_UNAUTHORIZED: false
+      WEBSOCKETS_ENABLED: true
+```
+
+Press the `esc` key to exit the insert mode and press `:` followed along with `wq` to quite the `vim` editor
+
+To ensure that Directus can write and save data in the `extensions` and `uploads` directories, enter the following command to grant the current user ownership of the directory:
+
+```bash
+sudo chown $USER:$USER extensions uploads
+```
+
+Run the command `docker-compose up`, which should spin up a Directus instance on port `80` of your EC2 server.
+
+Head to the EC2 dashboard and click on the instance ID; this will open the properties of your EC2 instance; here, you'll find the `Public IPv4 DNS.`
+
+![An EC2 instance properties page](ec2-public.png)
+
+Open this URL and you should find Directus running.
+
+## Set up AWS s3 (optional)
+
+If you want to use AWS S3 for media storage, follow these additional steps:
+
+Search for s3 on the console or navigate to the [s3 page](https://s3.console.aws.amazon.com/s3/home) to create a new storage bucket.
+
+![An AWS s3 page with list of buckets and a button to create a new bucket](new-bucket.png)
+
+When creating a new bucket, ensure ACLs are disabled and public access is blocked for the bucket for privacy (you can update these settings to suit your needs).
+
+![An AWS s3 bucket creation page with options to configure the bucket](create-bucket.png)
+
+Copy the name of the bucket and use it in your `docker-compose.yml` with the following configs:
+
+```yml
+  STORAGE_LOCATIONS: s3
+  STORAGE_S3_DRIVER: s3
+  STORAGE_S3_KEY: YOUR_AWS_KEY
+  STORAGE_S3_SECRET: YOUR_AWS_SECRET_KEY
+  STORAGE_S3_BUCKET: YOUR_STORAGE_NAME
+  STORAGE_S3_REGION: YOUR_PREFERRED_REGION
+  STORAGE_S3_ENDPOINT: s3.amazonaws.com
+```
+
+Re-run the Directus server with `docker-compose up` to implement the new changes added to the `docker-compose.yml` file.
+
+Alternatively, you can run the command `docker-compose up -d` to run the Directus application on the background
+
+## Next Steps
+
+Some steps you could consider moving forward towards improving your deployment include:
+
+- Improved security permissions for connecting to the created resources
+
+- Consider utilizing [AWS Secrets Manager](https://us-east-1.console.aws.amazon.com/secretsmanager/) for managing your `ENV` variables
+- Consider implementing a load balancer for scaling the incoming requests on your directus application.
```

Comment:
And it might be worth noting the limitations here against Cloud in a reasonable way: no backups, rolling updates, restart logic if it crashes.